beacon-chain/blockchain/process_block_test.go

```diff
@@ -1048,3 +1048,113 @@ func TestRemoveBlockAttestationsInPool_NonCanonical(t *testing.T) {
 	require.NoError(t, service.pruneCanonicalAttsFromPool(ctx, r, wrapper.WrappedPhase0SignedBeaconBlock(b)))
 	require.Equal(t, 1, service.cfg.AttPool.AggregatedAttestationCount())
 }
+
+func TestVerifyPandoraShardInfo(t *testing.T) {
+	ctx := context.Background()
+	beaconDB := testDB.SetupDB(t)
+
+	cfg := &Config{
+		BeaconDB: beaconDB,
+		StateGen: stategen.New(beaconDB),
+	}
+	service, err := NewService(ctx, cfg)
+	require.NoError(t, err)
+
+	s, err := v1.InitializeFromProto(&pb.BeaconState{Slot: 1, GenesisValidatorsRoot: params.BeaconConfig().ZeroHash[:]})
+	require.NoError(t, err)
+
+	genesisStateRoot := [32]byte{}
+	genesis := blocks.NewGenesisBlock(genesisStateRoot[:])
+	assert.NoError(t, beaconDB.SaveBlock(ctx, wrapper.WrappedPhase0SignedBeaconBlock(genesis)))
+	gRoot, err := genesis.Block.HashTreeRoot()
+	require.NoError(t, err)
+	service.finalizedCheckpt = &ethpb.Checkpoint{
```

Comment:
Private fields shouldn't be used in tests

Reply:
> Tests should be in a separate package

You are right. But this will require to refactor `prysm` implementation as well as ours - most of this private things are from `prysmaticlabs`. We are improving this client as a `vanguard` for our purpose now with no core modifications.

Reply:
You are right. But this will require to refactor prysm implementation as well as ours - most of this private things are from prysmaticlabs. We are improving this client as a vanguard for our purpose now with no core modifications. Good point, totally agree with this but now I can't refactor `prysm` codebase :(